File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:

        n = len(s)
        i=0
        j=n-1

        while i<j:
            while i<n and not s[i].isalnum():
                i+=1
            while j>=0 and not s[j].isalnum():
                j-=1
            if i>=j:
                break
                
            if s[i].isnumeric() and s[j].isnumeric():
                if s[i]!=s[j]:
                    return False
                pass
            elif s[i].lower() != s[j].lower():
                return False
            i+=1
            j-=1
        
        return True

File: 125-valid-palindrome/valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function (s) {
    function isAlNum(char) {
        return /^[a-zA-Z0-9]$/.test(char);
    }

    let temp = []
    for (let i = 0; i < s.length; i++) {
        if ((s[i] != " ")&(isAlNum(s[i]))) {
            temp.push(s[i].toLowerCase())
        }
    }
    return temp.join("") === temp.slice().reverse().join("")
};